android/src/com/frostwire/android/gui/activities/SettingsActivity.java

```diff
@@ -127,7 +127,7 @@ private void switchToFragment(String fragmentName, Bundle args, CharSequence tit
         Fragment f = Fragment.instantiate(this, fragmentName, args);
         FragmentTransaction transaction = getFragmentManager().beginTransaction();
         transaction.setTransition(FragmentTransaction.TRANSIT_FRAGMENT_FADE);
-        transaction.replace(R.id.activity_settings_content, f);
+        transaction.replace(R.id.activity_settings_content, f, fragmentName);
```

Comment:
so that the fragment has a tag and is found with `getFragments()` when `dispatchDialogClick` from in `AbstractDialog` is called

```
 if (activity instanceof AbstractActivity) {
            List<Fragment> fragments = ((AbstractActivity) activity).getFragments();  //here

            for (Fragment f : fragments) {
                dispatchDialogClickSafe(f, tag, which);
            }
        }
```